src/mod_muc_room.erl

```diff
@@ -156,9 +156,14 @@ init([Host, ServerHost, Access, Room, HistorySize, RoomShaper, Opts, QueueType])
     add_to_log(room_existence, started, NewState),
     {ok, normal_state, NewState}.
 
+%% This is a top level call to normal_state to kill the muc room during any event.
+%% Another hack we should remove once we have the process that sweeps empty mucs
+normal_state(Event, StateData) ->
+	{_, _, NewStateData} = normal_state(Event, StateData, true),
+	close_room_without_occupants(NewStateData).
```

Comment:
So is this essentially wrapping the call to normal_state so that we also call close_room_without_occupants after its called?

Reply:
And if so, why do this as opposed to calling close_room_without_occupants at the end of the existing normal_state method?

Reply:
There are like 5 or 6 normal_state methods, and some don't save their state in a variable and simply return it.  So its easier to capture it here adjust it.